app/queries/articles/active_threads_query.rb

```diff
@@ -0,0 +1,32 @@
+module Articles
+  class ActiveThreadsQuery
+    DEFAULT_OPTIONS = {
+      tags: ["discuss"],
+      time_ago: nil,
+      count: 10
+    }.with_indifferent_access.freeze
+
+    MINIMUM_SCORE = -4
+
+    def self.call(relation: Article.published, options: {})
+      options = DEFAULT_OPTIONS.merge(options)
+      tags, time_ago, count = options.values_at(:tags, :time_ago, :count)
+
+      relation.limit(count)
+      relation = if time_ago == "latest"
+                   relation.order(published_at: :desc).where(score: MINIMUM_SCORE..).presence ||
+                     relation.order(published_at: :desc)
+                 elsif time_ago
+                   relation.order(comments_count: :desc)
+                     .where(published_at: time_ago.., score: MINIMUM_SCORE..).presence ||
+                     relation.order(comments_count: :desc)
+                 else
+                   relation.order("last_comment_at DESC NULLS LAST")
```

Comment:
```
relation.order("last_comment_at DESC NULLS LAST")
```
It is possible that `nil` values end up in the final array containing all of the articles to be displayed in the sidebar. That said, it is necessary that we ensure that any `null` values are last in that array, otherwise the articles containing `nil` will be returned _before_ the articles not containing `nil`. The ordering here ensures that the relation is sorted properly and is tested on line `55` in the `active_threads_query_spec.rb`.

Reply:
great!

Reply:
Random tidbit: the Arel version shipping in Rails 6.1 supports `nulls_last` and `nulls_first`.